### ISSUE TITLE
Can't select namespace when pressing Enter

### ISSUE THREAD

**Issue by colltoaction** [kubernetes/dashboard#6227]
I think the issue arises from this line of code:

https://github.com/kubernetes/dashboard/blob/ac0fd117252caf4a17cab50dabe0f3b6cb8630eb/src/app/frontend/common/components/namespace/component.ts#L114

where the `value of the input` is taken, and I think the `select value` should be used.

![image](https://user-images.githubusercontent.com/1548532/124311417-1adecc00-db44-11eb-9fcc-ebc20b802e77.png)

version: `Dashboard v2.0.5+0.g8b89a1c7b`

**Comment by maciaszczykm** [comment]
I've tested it and this is actually working as intended. When you have namespaces like `abc`, `test` and `dev` listed and `ab` that you have access to but it is not listed the only option to get to it is to type `ab` and press enter. In that case we have to accept what user has typed instead of acting like autocompletion and redirecting to `abc`.

**Comment by maciaszczykm** [comment]
tl;dr: This feature allows manually entering the namaspace name. This is not pure autocompletion. The options below are here for convinience.

**Comment by floreks** [comment]
Exactly, filtering was added for convenience only, and changing the namespace on clicking enter is what we wanted to achieve. In case the user does not have privileges to list any namespace it is still possible to change the namespace.

**Comment by colltoaction** [comment]
I see. so when I use up and down arrows to navigate the filtered list I can
only click to select. and pressing Enter always picks the value of the
textbox. got it!
